includes/admin.form.inc

```diff
@@ -351,22 +354,1176 @@ function islandora_scholar_admin_pdf_form($form, &$form_state) {
+      'callback' => 'islandora_scholar_xpaths_citeproc_callback',
+      'wrapper' => 'islandora_scholar_xpaths_citeproc_title',
+    ),
+   '#field_suffix' => (!empty($form_state['values']['islandora_scholar_xpaths_demo']) ? _mods_id($form_state['values']['islandora_scholar_xpaths_role_term'], check_plain($form_state['values']['islandora_scholar_xpaths_demo'])) : t("No PID entered to verify MODS validity.")),
+  ));
+  module_load_include('inc', 'islandora', 'includes/solution_packs');
+  return system_settings_form($form);
+}
+
+/**
+ * Admin form validate function.
+ */
+function islandora_scholar_admin_pdf_form_validate($form, &$form_state) {
+  if ($form_state['values']['islandora_scholar_create_fulltext']) {
+    $islandora_scholar_path_to_pdftotext = $form_state['values']['islandora_scholar_path_to_pdftotext'];
+    exec($islandora_scholar_path_to_pdftotext, $output, $return_value);
+    if ($return_value != 99) {
+      form_set_error('islandora_scholar_path_to_pdftotext', t('Cannot extract text from PDF without a valid path to pdftotext.'));
+    }
+  }
+}
+
+/**
+ * Function updates PDF to text URL div.
+ */
+function islandora_scholar_update_pdftotext_url_div($form, $form_state) {
+  $form_state['rebuild'] = TRUE;
+  return $form['islandora_scholar_url_fieldset']['wrapper'];
+}
+
+/**
+ * Function removes and resets the custom XPaths.
+ */
+function _islandora_scholar_xpaths_reset() {
+  $islandora_scholar_xpaths = islandora_scholar_xpaths_list();
+  $message = t('All custom Islandora Scholar XPaths have been reset.');
+  drupal_set_message(check_plain($message));
+  array_walk($islandora_scholar_xpaths, 'variable_del');
+}
+
+/**
+ * Function specifies the array of intputs.
+ */
+function islandora_scholar_xpaths_list() {
+  $islandora_scholar_xpaths = array(
+    'islandora_scholar_xpaths_abstract',
+    'islandora_scholar_xpaths_authors_xpath',
+    'islandora_scholar_xpaths_call-number',
+    'islandora_scholar_xpaths_collection-title',
+    'islandora_scholar_xpaths_container-title',
+    'islandora_scholar_xpaths_created_date',
+    'islandora_scholar_xpaths_dateCaptured',
+    'islandora_scholar_xpaths_date_issued',
+    'islandora_scholar_xpaths_degree_grantor',
+    'islandora_scholar_xpaths_department',
+    'islandora_scholar_xpaths_DOI',
+    'islandora_scholar_xpaths_edition',
+    'islandora_scholar_xpaths_embargo_date',
+    'islandora_scholar_xpaths_enabled',
+    'islandora_scholar_xpaths_end_page',
+    'islandora_scholar_xpaths_event',
+    'islandora_scholar_xpaths_event_place',
+    'islandora_scholar_xpaths_event_place_property',
+    'islandora_scholar_xpaths_event_property',
+    'islandora_scholar_xpaths_family',
+    'islandora_scholar_xpaths_genre',
+    'islandora_scholar_xpaths_given',
+    'islandora_scholar_xpaths_host_title',
+    'islandora_scholar_xpaths_ISBN',
+    'islandora_scholar_xpaths_issn',
+    'islandora_scholar_xpaths_issue',
+    'islandora_scholar_xpaths_local_author',
+    'islando
```

Comment:
Just spotted by coincidence: some HTML seems to have sneaked in here... ;)

Reply:
Thanks, I'll correct this. Wow completely missed that.